server/controllers/userAuth.js

```diff
@@ -126,9 +126,70 @@ const signUp = (req, resp, next) => {
 
 const confirmEmail = (req, resp) => {};
 
+const resetPassword = (req, resp, next) => {
+  const { email } = req.body;
+  const sanitizedEmail = sanitize(email);
+  const { isEmail } = validator;
+
+  if (!isEmail(sanitizedEmail)) {
+    return resp.sendStatus(406);
+  }
+
+  User.findOne({ email: sanitizedEmail })
+    .exec()
+    .then(user => {
+      if (!user) {
+        throw new Error();
+      }
+
+      const { displayName, isActive, idFromProvider } = user;
+
+      if (idFromProvider) {
+        throw new Error('authorization.actions.reset');
+      }
+
+      if (!isActive) {
+        throw new Error(
+          'authorization.actions.reset-password-not-active-account'
+        );
+      }
+
+      const resetToken = crypto.randomBytes(32).toString('hex');
+      const resetTokenExpirationDate = new Date().getTime() + 3600000;
+
+      return User.findOneAndUpdate(
```

Comment:
do you need to search for this document again? Are you using unsanitized email?

Reply:
yes, because I have to update User model properties.

Sanitized email fixed.